llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -1126,6 +1125,28 @@ def HasStdExtZbbOrZbkbOrP
                          "'Zbkb' (Bitmanip instructions for Cryptography) or "
                          "'Base P' (Packed-SIMD)">;
 
+// "Y" extension (CHERI support)
+
+def FeatureStdExtY : RISCVExperimentalExtension<0, 96, "'Base Y' (CHERI)">;
+def HasStdExtY
+    : Predicate<"Subtarget->hasStdExtY()">,
+      AssemblerPredicate<(all_of FeatureStdExtY), "'Base Y' (CHERI)">;
+
+// When enabled all memory operations (e.g. loads/stores) uses capability
+// registers as the base operand instead of the address sub-register.
+// Currently, capability mode needs to be chosen at assembly time, but follow-up
+// commits will add support for "hybrid" mode that adds instructions to
+// dynamically switch between capability mode and address mode (the latter being
+// fully backwards compatible with non-Y code).
+def FeatureCapMode : SubtargetFeature<"cap-mode", "IsCapMode", "true",
+                                      "Capability pointer mode">;
+def IsCapMode
```

Comment:
We renamed this to `XCheriPureCap` using `RISCVExtension`, so that it plays nice with the rest of the RISCV extension infrastructure. That enables us to do things like have `XCheriot` automatically imply `XCheriPureCap`.

Reply:
Though given the direction of the Y standard, perhaps we should invert the sense of the feature bit?